server/application/rest/service_shifts.py

```diff
@@ -10,42 +12,65 @@
 from application.rest.work_shift import HTTP_STATUS_CODES_MAPPING
 from responses import ResponseTypes
 from serializers.service_shift import ServiceShiftJsonEncoder
-import json
 
-service_shift_bp = Blueprint('service_shift', __name__)
+service_shift_bp = Blueprint("service_shift", __name__)
+
 
-@service_shift_bp.route('/service_shift', methods=['GET', 'POST'])
+@service_shift_bp.route("/service_shift", methods=["GET", "POST"])
 @cross_origin()
-def service_shift():
+def handle_service_shift():
     """
-    Handles POST to add service shift, GET to list all shifts for a shelter.
+    Handles POST to add service shifts and GET to list all shifts for a shelter.
     """
     repo = ServiceShiftsMongoRepo()
-    if request.method == 'GET':
-        shelter_id_string = request.args.get('shelter_id')
-        shelter_id = int(shelter_id_string) if shelter_id_string else None
-        shifts_as_dict = service_shifts_list_use_case(repo, shelter_id)
-        shifts_as_json = [
-            json.dumps(service_shift, cls = ServiceShiftJsonEncoder)
-            for service_shift in shifts_as_dict
-        ]
+
+    if request.method == "GET":
+        shelter_id_str = request.args.get("shelter_id")
+        filter_start_after_str = request.args.get("filter_start_after")
+
+        # Ensure proper conversion to int, handling empty or invalid cases
+        shelter_id = int(shelter_id_str) if shelter_id_str and shelter_id_str.isdigit() else None
+        filter_start_after = (
+            int(filter_start_after_str) if filter_start_after_str and filter_start_after_str.isdigit() else None
+        )
+
+        shifts = service_shifts_list_use_case(repo, shelter_id, filter_start_after)
+
         return Response(
-            shifts_as_json,
-            mimetype='application/json',
-            status=HTTP_STATUS_CODES_MAPPING[ResponseTypes.SUCCESS]
+            json.dumps(shifts, cls=ServiceShiftJsonEncoder),
+            mimetype="application/json",
+            status=HTTP_STATUS_CODES_MAPPING[ResponseTypes.SUCCESS],
         )
-    elif request.method == 'POST':
+
+    if request.method == "POST":
         shifts_as_dict = request.get_json()
-        print(shifts_as_dict)
-        # convert to ServiceShift objects
-        shifts_obj = [ServiceShift.from_dict(shift) for shift in shifts_as_dict]
+
+        # Validate JSON payload
+        if not shifts_as_dict:
+            return Response(
+                json.dumps({"message": "Invalid JSON object"}),
+                mimetype="application/json",
+                status=HTTP_STATUS_CODES_MAPPING[ResponseTypes.PARAMETER_ERROR],
+            )
+
+        try:
+            shifts_obj = [ServiceShift.from_dict(shift) for shift in shifts_as_dict]
```

Comment:
```suggestion
            shifts_obj = [
                ServiceShift.from_dict(shift)
                for shift in shifts_as_dict
            ]
```
This should resolve the 4th linting error